tests/test_shardformer/test_model/_utils.py

```diff
@@ -141,10 +142,16 @@ def _criterion(outputs, inputs):
     data = data_gen_fn()
     sharded_model.train()
     if booster.plugin.stage_manager is not None:
-        data = {
-            k: v.to('cuda').repeat(4, 1) if torch.is_tensor(v) or 'Tensor' in v.__class__.__name__ else v
-            for k, v in data.items()
-        }
+        if org_model.__class__.__name__ == 'BertForMultipleChoice':
+            for k, v in data.items():
+                repeat_size = [1] * v.dim()
+                repeat_size[0] = 4
+                data[k] = v.repeat(*repeat_size).to('cuda')
```

Comment:
Can you modify date generation function instead of hardcode here?

Reply:
modified